### ISSUE TITLE
Training loss might not be summed

### ISSUE THREAD

**Issue by morawi** [creafz/pytorch-cnn-finetune#1]
I think the training loss is not accumulated over all the batches, not sure if I am missing something here.
# train_loss added below
``
```
def train(epoch):
    model.train()
    train_loss=0
    for batch_idx, (data, target) in enumerate(train_loader):
        if args.cuda:
            data, target = data.cuda(), target.cuda()
        data, target = Variable(data), Variable(target)
        optimizer.zero_grad()
        output = model(data)
        loss = criterion(output, target)
        loss.backward()
        optimizer.step()
        train_loss +=loss.data[0]
        if batch_idx % args.log_interval == 0:
            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
                epoch, batch_idx * len(data), len(train_loader.dataset),
                100. * batch_idx / len(train_loader), train_loss/(batch_idx+1) ))

```
`
`

**Comment by creafz** [comment]
You are right, `train` function prints a loss value for every nth batch, where n is determined by the `--log-interval` argument, so by default, the function will print a loss for batch #100, batch #200 and so on. I will change the code to print an average loss instead of a loss for a particular batch.

**Comment by creafz** [comment]
I have changed the example, now `train` function prints average loss per element. I have chosen per element loss and not per batch loss to preserve the same behaviour for both `train` and `test` functions)
